### ISSUE TITLE
[Bug]: import of classes is broken for 0.6.1 in vue

### ISSUE THREAD

**Issue by jesus-chacon** [web-infra-dev/rsbuild#2073]
### Version

```sh
System:
    OS: macOS 13.6
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1003.39 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 123.0.6312.123
    Safari: 16.6
  npmPackages:
    @rsbuild/core: 0.6.1 => 0.6.1
    @rsbuild/plugin-image-compress: 0.6.1 => 0.6.1
    @rsbuild/plugin-lightningcss: 0.6.1 => 0.6.1
    @rsbuild/plugin-vue: 0.6.1 => 0.6.1
```

### Details

I have update from 0.6.0 (in this version the app work properly) this morning I have updated to 0.6.1 and right now in the browser I have seen this error.

`Error loading stores ReferenceError: TherapistApiConnector is not defined
    at new SetTherapistActionHandler (SetTherapist.ts:4:59)
    at Proxy.setTherapist (Store.ts:35:41)
    at Proxy.<anonymous> (pinia.mjs:1380:30)
    at store.<computed> (pinia.mjs:935:50)
    at Proxy.initStores (LoadStoresData.vue:63:39)
    at async TcSuspense.vue:36:7`

TherapistApiConnector is a class for wrap the api request. I migrate this class to a functional implementation and it work properly. I can build the code without errors but when you load the web receive the same errors. I have test it in several environments (Jenkins, mac, ubuntu and windows) and I have received the same result.

### Reproduce link

https://github.com/jesus-chacon/rspack_class_error

### Reproduce Steps

yarn install

yarn dev

Open the browser go to http://localhost:3000

I only implement a basic store and request flow for sandbox. You can downgrade the version to 0.6.0 and check that all work properly.

**Comment by jesus-chacon** [comment]
Sorry to be late. I have update the link for check the error I only implement a basic flow with the default template in vue. Thanks for help.

**Comment by jesus-chacon** [comment]
@fi3ework in 0.6.3 it work properly again

**Comment by chenjiahan** [comment]
I can not reproduce the error too, and it seems that this issue has been fixed by Rspack v0.6.2.

So I will close this issue, feel free to reopen it if the error occurs again.
